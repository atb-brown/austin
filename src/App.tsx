import React from 'react';
import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Austin's React website.<br />
          <a className='App-link' href="https://github.com/atb-brown/austin">(see the source code)</a>
        </p>
        <p text-align="right">
          <h3>
            Links
          </h3>
          <div>
            <div>
              GitHub
            </div>
            <div>
              <a className='App-link' href='https://github.com/atb-brown'>atb-brown</a>
            </div>
            <br />
          </div>
          <div>
            <div>
              LinkedIn
            </div>
            <div>
              <a className='App-link' href='https://www.linkedin.com/in/austin-brown-06382a12a/'>austin-brown-06382a12a</a>
            </div>
            <br />
          </div>
          <div>
            <div>
              Agile Alliance Scrum Master Certification
            </div>
            <div>
              <a className='App-link' href='https://bcert.me/sixkczuof'>sixkczuof</a>
            </div>
            <div className='App-soft-text'>
              Issued by: <a className='App-soft-text' href='https://www.scrumalliance.org/'>Scrum Alliance, Inc.</a>
            </div>
            <br />
          </div>
          <div>
            <div>
              SAFe 5 Scrum Master Certification
            </div>
            <div>
              <a className='App-link' href='https://www.credly.com/badges/18e3c923-956c-4525-b857-5f9c0dfb9713/public_url'>18e3c923-956c-4525-b857-5f9c0dfb9713</a>
            </div>
            <div className='App-soft-text'>
              Issued by: <a className='App-soft-text' href='https://scaledagile.com/'>Scrum Alliance, Inc.</a>
            </div>
            <br />
          </div>
        </p>
      </header>
    </div>
  );
}

export default App;
