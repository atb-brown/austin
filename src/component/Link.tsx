import React from "react";

/**
 * A basic, clickable link for a URL.
 *
 * @param {LinkProps} props Properties.
 * @return {ReactElement} The link component.
 */
export default function Link(props: LinkProps) {
  return (
    <a className="App-link" href={props.url}>
      {props.description}
    </a>
  );
}

type LinkProps = {
  description: string;
  url: string;
};
