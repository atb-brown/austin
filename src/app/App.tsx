import Stats from "../component/widget/statistic/Stats";
import useWindowDimensions from "../hook/dimension/useWindowDimensions";
import "./App.css";
import AppAcknowledgements from "./AppAcknowledgements";
import AppBody, { AppBodyProviders } from "./AppBody";
import AppTitle from "./AppTitle";
import React from "react";

// TODO: Come up with a more elegant solution for injection.
export type AppProviders = AppBodyProviders; // ... & OtherComponentProviders & AnotherComponentProviders; <- union with other providers, if the need arises.

/**
 * The main component for the site application.
 *
 * @return {ReactElement} React Component
 */
function App({ guestProvider }: AppProviders) {
  const { width, height } = useWindowDimensions();
  return (
    <div
      className="App"
      style={{
        minWidth: width,
        maxWidth: width,
        minHeight: height,
        maxHeight: height,
      }}
    >
      <header className="App-header">
        <Stats />
        <AppTitle />
        <AppBody guestProvider={guestProvider} />
        <AppAcknowledgements />
      </header>
    </div>
  );
}

export default App;
