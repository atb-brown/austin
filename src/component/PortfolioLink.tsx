import Link from "./Link";
import React from "react";

export default function PortfolioLink(props: LinkProps) {
  if (props.issuerDescription && props.issuerUrl) {
    return (
      <LinkWithIssuer
        description={props.description}
        url={props.url}
        urlId={props.urlId}
        issuerDescription={props.issuerDescription}
        issuerUrl={props.issuerUrl}
      />
    );
  }
  return (
    <LinkWithoutIssuer
      description={props.description}
      url={props.url}
      urlId={props.urlId}
    />
  );
}

function LinkWithoutIssuer(props: LinkProps) {
  // TODO: it's a bad idea to have <div/> elements that are children of <p/> elements
  return (
    <div>
      <div>{props.description}</div>
      <div>
        <Link description={props.urlId} url={props.url} />
      </div>
      <br />
    </div>
  );
}

function LinkWithIssuer(props: LinkProps) {
  return (
    <div>
      <div>{props.description}</div>
      <div>
        <Link description={props.urlId} url={props.url} />
      </div>
      <div className="App-soft-text">
        Issued by:{" "}
        <Link
          description={props.issuerDescription ?? ""}
          url={props.issuerUrl ?? ""}
        />
      </div>
      <br />
    </div>
  );
}

type LinkProps = {
  description: string;
  url: string;
  urlId: string;
  issuerDescription?: string;
  issuerUrl?: string;
};
