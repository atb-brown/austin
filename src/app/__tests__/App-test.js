import { wait as guestBook } from "../../test/component/GuestBookTestUtils";
import { registerMockProvider as mockGuestBookProvider } from "../../test/mock/hook/mockUseGuestBookInfo";
import App from "../App";
import { render } from "@testing-library/react";

it("renders to match snapshot", async () => {
  mockGuestBookProvider();
  const renderResult = render(<App />);

  await guestBook(renderResult);

  expect(renderResult.asFragment()).toMatchSnapshot();
});
