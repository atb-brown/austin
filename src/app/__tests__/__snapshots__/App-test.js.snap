// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
    style="max-height: 768px; max-width: 1024px; min-height: 768px; min-width: 1024px;"
  >
    <header
      class="App-header"
    >
      <div
        class="Stats Stats-Visual Stats-Container clickable"
        data-testid="stats-ribbon"
      >
        <div
          class="Stats"
          style="width: 10%;"
        >
          <div>
             | 
          </div>
          <div
            style="display: flex; flex-direction: column; font-size: 50%;"
          >
            <div>
              Browser
            </div>
            <div>
              WebKit
            </div>
          </div>
          <div>
             | 
          </div>
          <div
            style="display: flex; flex-direction: column; font-size: 40%;"
          >
            <div>
              Window Size
            </div>
            <div>
              1024 x 768
            </div>
          </div>
        </div>
      </div>
      <div
        style="display: flex; width: 100%;"
      >
        <img
          alt="logo"
          class="App-logo disable-text-selection"
          src="[object Object]"
        />
        <div>
          Austin's React Website
          <br />
          <a
            class="App-link"
            href="https://github.com/atb-brown/austin"
          >
            (see the source code)
          </a>
        </div>
      </div>
      <div
        style="display: flex; width: 100%;"
      >
        <div
          style="display: flex; flex: 1;"
        >
          <div
            style="align-items: center; display: flex; flex: 1; flex-direction: column;"
          >
            <h1>
              Links
            </h1>
            <div
              class="App-text"
            >
              <div>
                GitHub
              </div>
              <div>
                <a
                  class="App-link"
                  href="https://github.com/atb-brown"
                >
                  atb-brown
                </a>
              </div>
              <br />
            </div>
            <div
              class="App-text"
            >
              <div>
                StackOverflow
              </div>
              <div>
                <a
                  class="App-link"
                  href="https://stackoverflow.com/users/5476186/austin-brown"
                >
                  5476186/austin-brown
                </a>
              </div>
              <br />
            </div>
            <div
              class="App-text"
            >
              <div>
                LinkedIn
              </div>
              <div>
                <a
                  class="App-link"
                  href="https://www.linkedin.com/in/austin-brown-06382a12a/"
                >
                  austin-brown-06382a12a
                </a>
              </div>
              <br />
            </div>
            <div
              class="App-text"
            >
              <div>
                Scrum Master Certification
              </div>
              <div>
                <a
                  class="App-link"
                  href="https://bcert.me/sixkczuof"
                >
                  sixkczuof
                </a>
              </div>
              <div
                class="App-soft-text"
              >
                Issued by: 
                <a
                  class="App-link"
                  href="https://www.scrumalliance.org/"
                >
                  Scrum Alliance, Inc.
                </a>
              </div>
              <br />
            </div>
            <div
              class="App-text"
            >
              <div>
                SAFe 5 Scrum Master Certification
              </div>
              <div>
                <a
                  class="App-link"
                  href="https://www.credly.com/badges/18e3c923-956c-4525-b857-5f9c0dfb9713/public_url"
                >
                  18e3c923-956c-4525-b857-5f9c0dfb9713
                </a>
              </div>
              <div
                class="App-soft-text"
              >
                Issued by: 
                <a
                  class="App-link"
                  href="https://scaledagile.com/"
                >
                  Scaled Agile, Inc.
                </a>
              </div>
              <br />
            </div>
          </div>
        </div>
        <div
          style="display: flex; flex: 3;"
        >
          <div
            style="align-items: center; display: flex; flex: 1; flex-direction: column; gap: 10px; justify-content: center; width: 100%;"
          >
            <div
              style="align-items: center; display: flex; flex: 1; flex-direction: row; justify-content: space-evenly; width: 100%;"
            >
              <div
                class="clickable"
                data-testid="remember-me"
                style="align-items: center; border-radius: 10px; display: flex; font-size: 50%; height: 60px; justify-content: center; width: 25%;"
                title="This component allows the site to remember a user. It's an exercise in handling cookies and component state!"
              >
                Click me if you want me to remember you!
              </div>
              <div
                class="disable-text-selection clickable"
                data-testid="counter"
                style="align-items: center; border-radius: 10px; display: flex; font-size: 20px; height: 60px; justify-content: center; width: 25%;"
                title="This component counts the number of times it has been clicked. It's an exercise in handling clicks, component state, and cookie management!"
              >
                0
              </div>
            </div>
            <div
              style="align-items: center; display: flex; flex: 8; flex-direction: column; justify-content: center; width: 95%;"
            >
              <div
                data-testid="guest-book"
                style="display: flex; flex-direction: column; gap: 3px; min-height: 50%; width: 99%;"
                title="This component displays everyone who has signed my guest book. It's an exercise in HTTP requests and dependency injection!"
              >
                <div
                  style="border-bottom-left-radius: 15px; border-top-left-radius: 15px; border-top-right-radius: 15px; display: flex; height: 85px; overflow: hidden; width: 100%;"
                >
                  <div
                    class="Guests-Visual"
                    style="flex: 2; flex-direction: column; justify-content: space-evenly;"
                  >
                    <div
                      class="Guests-Text-container"
                      style="font-size: 75%;"
                    >
                      austin
                    </div>
                    <div
                      class="Guests-Text-container"
                      style="font-size: 50%; justify-content: center;"
                    >
                      <a
                        href="www.example.com"
                      >
                        www.example.com
                      </a>
                    </div>
                  </div>
                  <div
                    class="Guests-Visual Guests-Text-Container"
                    style="flex: 7; margin-left: 3px; margin-right: 3px;"
                  >
                    <div
                      class="Guests-Text-Container"
                      style="font-size: 75%; height: 100%; width: 98%;"
                    >
                      info loaded
                    </div>
                  </div>
                  <div
                    class="Guests-Visual Guests-Text-Container"
                    style="flex: 1; font-size: 60%;"
                  >
                    2023
                  </div>
                </div>
                <a
                  class="clickable"
                  href="https://github.com/atb-brown/guest-book/edit/main/guests.json"
                  style="align-items: center; align-self: flex-end; border-bottom-left-radius: 15px; border-bottom-right-radius: 15px; border-top-left-radius: 15px; color: white; display: flex; font-size: 80%; height: 85px; justify-content: center; text-decoration: none; width: 15%;"
                >
                  Sign my guest book!
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="align-items: flex-end; display: flex; flex-direction: column; justify-content: flex-end; margin-right: 3%; width: 100%;"
      >
        <div
          class="App-soft-text"
          style="font-size: 20px; padding-left: 50px;"
        >
          <a
            class="App-link"
            href="https://github.com/atb-brown/austin/commit/5c8514a96194e6271d4e8e7f234499ee4e95ab3d"
          >
            Initialized
          </a>
           using 
          <a
            class="App-link"
            href="https://create-react-app.dev/"
          >
            create-react-app
          </a>
           
        </div>
        <div
          class="App-soft-text"
          style="font-size: 20px;"
        >
          This website is deployed by borrowing the incredible efforts of the 
          <a
            class="App-link"
            href="https://github.com/gitname/react-gh-pages/graphs/contributors"
          >
            react-gh-pages contributors
          </a>
          .
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
